### PR TITLE
feat: add simple preprocessing for nested documents

### PR DESCRIPTION
**Pull request by violenil**
Preprocessing should look as follows:

- [x] processed target Documents (indexing) contain only one image and one text chunk
- [x] processed query Documents contain one text chunk

### CODE REVIEW

File: now_common/utils.py [hunk 1]
```diff
@@ -208,6 +209,28 @@ def _get_sentence_docs(batch):
...
+    Process a `DocumentArray` with `Document`s that have `chunks` of nested `Document`s.
+    It constructs `Document`s containg two chunks: one containing image data and another
+    containing text data. Fields for indexing should be specified in the `UserInput`.
+
+    :param da: A `DocumentArray` containing nested chunks.
+    :return: A `DocumentArray` with `Document`s containing text and image chunks.
+    """
+    fields = user_input.task.indexer_scope
+    es_transformer = ESDataTransformer()
+    transformed_da = es_transformer.transform(da)
```
Comment (jupyterjazz) [inline now_common/utils.py:223]:
```suggestion
    transformed_da = ESDataTransformer.transform(da)
```

File: now_common/utils.py [hunk 2]
```diff
@@ -208,6 +209,28 @@ def _get_sentence_docs(batch):
...
+    processed_da = DocumentArray()
+    for doc in transformed_da:
+        new_doc = Document(chunks=
+            [
+                Document(text=doc[fields['text']][0]),
+                Document(uri=doc[fields['image']][0])
+            ]
+        )
+        processed_da.append(new_doc)
+    return processed_da
```
Comment (jupyterjazz) [inline now_common/utils.py:233]:
```suggestion
    return DocumentArray([Document(chunks=
        [
            Document(text=doc[fields['text']][0]),
            Document(uri=doc[fields['image']][0])
        ]
    ) for doc in transformed_da])
```

File: tests/unit/data_loading/test_data_loading.py [hunk 3]
```diff
@@ -124,6 +125,20 @@ def test_da_custom_ds(da: DocumentArray):
     for doc in loaded_da:
         assert doc.blob != b''
 
+def test_es_custom_ds(es_resource_path: str = 'es_extracted_data_flat'):
```
Comment (jupyterjazz) [inline tests/unit/data_loading/test_data_loading.py:128]: can you upload the data?
also, what is this test checking by `assert doc.blob != b''`?

**Review by jupyterjazz — COMMENTED** [review]
Nice!
left minor questions/suggestions

File: tests/unit/data_loading/test_data_loading.py (hunk 3 above)
Comment (violenil) [reply tests/unit/data_loading/test_data_loading.py:129]: > can you upload the data?

I have not got a go-ahead from Max to upload this data yet.

> also, what is this test checking by assert doc.blob != b''?

make sure that the doc is not empty :)

**Pull request merged**